src/StackExchange.Redis/Interfaces/IDatabase.cs

```diff
@@ -721,6 +721,31 @@ public interface IDatabase : IRedis, IDatabaseAsync
         /// <remarks>https://redis.io/commands/lpop</remarks>
         RedisValue[] ListLeftPop(RedisKey key, long count, CommandFlags flags = CommandFlags.None);
 
+        /// <summary>
+        /// Scans through the list stored at <paramref name="key"/> looking for <paramref name="element"/>, returning the 0-based
+        /// index of the first matching element.
+        /// </summary>
+        /// <param name="key">The key of the list.</param>
+        /// <param name="element">The element to search for.</param>
+        /// <param name="rank">The rank of the first element to return, within the sub-list of matching indexes in the case of multiple matches.</param>
+        /// <param name="maxLength">The maximum number of elements to scan through before stopping, defaults to 0 (a full scan of the list.)</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>The 0-based index of the first matching element, null if not found.</returns>
+        long? ListPosition(RedisKey key, RedisValue element, long rank = 1, long maxLength = 0, CommandFlags flags = CommandFlags.None);
+
+        /// <summary>
+        /// Scans through the list stored at <paramref name="key"/> looking for <paramref name="count"/> instances of <paramref name="element"/>, returning the 0-based
+        /// indexes of the the matching elements.
+        /// </summary>
+        /// <param name="key">The key of the list.</param>
+        /// <param name="element">The element to search for.</param>
+        /// <param name="count">The number of matches to find. A count of 0 will return the indexes of all occurrences of the element.</param>
+        /// <param name="rank">The rank of the first element to return, within the sub-list of matching indexes in the case of multiple matches.</param>
+        /// <param name="maxLength">The maximum number of elements to scan through before stopping, defaults to 0 (a full scan of the list.)</param>
+        /// <param name="flags">The flags to use for this operation.</param>
+        /// <returns>An array of at most <paramref name="count"/> of indexes of matching elements. If none are found, and empty array is returned</returns>
+        RedisValue[] ListPositions(RedisKey key, RedisValue element, long count, long rank = 1, long maxLength = 0, CommandFlags flags = CommandFlags.None);
```

Comment:
I'm confused on intent here - looking at the API, I'd expect a `long[]` here, with `-1` for not found (same for `ListPosition`) with `-1` being not found. For example `ListInsertAfter`/`ListInsertBefore` or any of the `Array` operations in .NET. Thoughts?

Reply:
@NickCraver The reasoning here is that the command returns `(nil)` if not found.

```sh
127.0.0.1:6379> LPUSH foo foo
(integer) 1
127.0.0.1:6379> LPOS foo foo
(integer) 0
127.0.0.1:6379> LPOS foo bar
(nil)
```

For Convention's sake, I think you're probably right, returning -1 makes more sense.

I did not have `long[]` as the return type because I hadn't seen any primitive-array return types in the interface, wasn't quite sure the best way to extract it, I can take a crack at adding the `ResultProcessor` to handle it, shouldn't be too difficult.

If the return from Redis is < `count` then can pad the array with `-1`? Right now it's just passing back an array where the number of values corresponds to the matches. (So if you request 6 and there are only 3 it will only return 3)

Thoughts?

Reply:
Ah gotcha - I'd expect `long` with `-1` ass a default for the singular case, and an empty array in the plural case - does that make sense?

Reply:
To save time, I think processor should be:
```cs
private sealed class LongArrayProcessor : ResultProcessor<long[]>
{
    protected override bool SetResultCore(PhysicalConnection connection, Message message, in RawResult result)
    {
        if (result.Type == ResultType.MultiBulk && !result.IsNull)
        {
            var arr = result.ToArray((in RawResult x) => (long)x.AsRedisValue())!;
            SetResult(message, arr);
            return true;
        }
        return false;
    }
}
```